### PR TITLE
Fix bug: cannot re-run helm uninstall if the first one failed and cannot fetch logs of failed uninstalltion job

### PR DESCRIPTION
**Pull request by PhanLe1010**
1. Adding `before-hook-creation` annotation value instruct Helm to remove the previous uninstallation job and before try to avoid the resource conflict https://helm.sh/docs/topics/charts_hooks/#:~:text=Description-,before%2Dhook%2Dcreation,-Delete%20the%20previous
2. Set `restartPolicy: Never` to keep the log of failed uninstallation pod so users can identify the reason that it is failing, see more https://kubernetes.io/docs/concepts/workloads/controllers/job/#:~:text=Note%3A%20If,not%20lost%20inadvertently.

longhorn/longhorn#4711

### CODE REVIEW

File: chart/templates/uninstall-job.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
-      restartPolicy: OnFailure
+      restartPolicy: Never
```
Comment (innobead) [inline chart/templates/uninstall-job.yaml:33]: `backoffLimit` has been configured, so is it possible `OnFailure` is intended somehow?

Comment (PhanLe1010) [reply chart/templates/uninstall-job.yaml:33]: backoffLimit is set to 1 so I don't see a difference between `restartPolicy: Never` and `restartPolicy: restartPolicy: Never` in terms of restarting the failed container (a.k.a if container failed only 1 time, stop the job)

Comment (PhanLe1010) [reply chart/templates/uninstall-job.yaml:33]: Also, user can always trigger `helm uninstall` again to retry so I am not too worry about `restartPolicy`

**Review by innobead — APPROVED** [review]

File: chart/templates/uninstall-job.yaml (hunk 1 above)
Comment (innobead) [reply chart/templates/uninstall-job.yaml:33]: LGTM!

**Pull request merged**